linux/LinuxProcessTable.c

```diff
@@ -141,20 +141,29 @@ static void LinuxProcessTable_initTtyDrivers(LinuxProcessTable* this) {
    if (r < 0)
       return;
 
-   int numDrivers = 0;
-   int allocd = 10;
+   size_t numDrivers = 0;
+   size_t allocd = 10;
    ttyDrivers = xMallocArray(allocd, sizeof(TtyDriver));
    char* at = buf;
-   while (*at != '\0') {
+   char* path = NULL;
+   while (at && *at != '\0') {
+      /*
+       * Format:
+       * [name]  [node path]  [major]  [minor range]  [type]
+       * serial  /dev/ttyS    4        64-95          serial
+       */
       at = strchr(at, ' ');    // skip first token
+      if (!at) goto finish;    // bail out on truncation
```

Comment:
Avoid single-line if statements.

Also maybe leave some space after each block to group the various segment of the parsed line …